src/components/select/demoSelectHeader/index.html

```diff
@@ -0,0 +1,22 @@
+<div ng-controller="SelectHeaderController" class="md-padding" ng-cloak>
+  <div>
+    <h1 class="md-title">Pick a vegetable below</h1>
+    <div layout="row">
+      <md-input-container>
+        <label>Vegetables</label>
+        <md-select ng-model="selectedVegetables" md-on-close="clearSearchTerm()" multiple>
+          <md-select-header class="demo-select-header">
+            <input ng-model="searchTerm"
+                   type="search"
+                   placeholder="Search for a vegetable.."
+                   class="demo-header-searchbox _md-text">
+          </md-select-header>
+          <md-optgroup label="vegetables">
+            <md-option ng-value="vegetable" ng-repeat="vegetable in vegetables |
```

Comment:
Hello Eric,

Solid point, I am actually working on #7385 which will mimic the `autocomplete` directive more closely and will also replace this code.

My intention for this demo was really just to give an example of how the `md-select-header` could be used. I hope it, in no way, serves as an example of what a dedicated search bar component would look like (but that is coming soon :+1: ).

This demo will change after the search component PR is finished.